packages/core/src/shared/globalState.ts

```diff
@@ -49,6 +49,7 @@ export type globalKey =
     | 'aws.toolkit.lsp.manifest'
     | 'aws.amazonq.customization.overrideV2'
     | 'aws.amazonq.regionProfiles'
+    | 'aws.amazonq.regionProfiles.cachedResult'
```

Comment:
why not set a flag on the items in `aws.amazonq.regionProfiles` ? why does this need an entirely separate structure?

Reply:
> 'aws.amazonq.regionProfiles'

this is to memorize previous selected option, so we will need a separate structure to cache previous api call response

Reply:
> > 'aws.amazonq.regionProfiles'
> 
> this is to memorize previous selected option

but that could be a flag on the item(s) in the structure. why is a separate structure needed?

Reply:
that makes sense, i will reuse the existing one then

Reply:
@justinmk3 i encounter an issue or say concern when i am doing this exercise https://github.com/aws/aws-toolkit-vscode/pull/7151/commits/3e501456fb1cd7b125819db2fbe7390cdfdc2558#diff-f5716036a70a8b212e9dae829ec09e33731f181c173a3f2b9ad9d361d51af14bR54-R56, if I want to reuse the same global key, i might need to restructure the schema, in such case, users who already selected profile update their plugin, they might need to re-select the profile again. Given the sev2s we have right now, I am concerned if we want to do this now or is there another way to deal with such scenario?

Reply:
Definitely want to avoid a restructure. If a restructure is required, then I guess we need a new globalState field.

Reply:
noted